test/jsdom.ts

```diff
@@ -0,0 +1,29 @@
+// @ts-expect-error - JSDOM types are missing
+import { JSDOM } from 'jsdom';
+
+type JSDOM = any;
+
+const jsdomGlobals: Record<string, (dom: JSDOM) => any> = {
+  Event: (jsdom: JSDOM) => jsdom.window.Event,
+  Node: (jsdom: JSDOM) => jsdom.window.Node,
+  window: (jsdom: JSDOM) => jsdom.window,
+  document: (jsdom: JSDOM) => jsdom.window.document,
+  requestAnimationFrame: (jsdom: JSDOM) => jsdom.window.requestAnimationFrame,
+};
+
+/** Setup DOM globals required by Preact rendering. */
+export function setupJSDOM() {
+  // Enable `requestAnimationFrame` which Preact uses for scheduling hooks.
+  const dom = new JSDOM('', { pretendToBeVisual: true });
+  const g = global as any;
```

Comment:
Hmm... should we be using `globalThis` here?

Reply:
Yea good call. globalThis was added in Node 12 so should be safe to use now. I've updated all references of `global` to `globalThis`.